source/RoslynAnalyzers/Extensions.cs

```diff
@@ -34,4 +35,54 @@ public static INamespaceSymbol GetTopMostNamespace(this INamespaceSymbol ns)
                 ? ns
                 : GetTopMostNamespace(ns.ContainingNamespace);
     }
+
+    public static class SyntaxExtensions
+    {
+        /// <summary>
+        /// Walks up the syntax tree until it reaches a namespace at the top. Returns emptystring if one cannot be found
+        /// </summary>
+        /// <remarks>
+        /// Adapted from https://andrewlock.net/creating-a-source-generator-part-5-finding-a-type-declarations-namespace-and-type-hierarchy/
+        /// </remarks>
+        public static string GetNamespace(this SyntaxNode syntax)
+        {
+            // If we don't have a namespace at all we'll return an empty string
+            // This accounts for the "default namespace" case
+            var nameSpace = string.Empty;
```

Comment:
Should we have a "marker" value here instead of `String.Empty`?

Reply:
🤷 this was third party code I got from a blog that I tweaked.
It seems like the right thing to do; If you just write a class in C# with no namespace, you literally get no namespace, not a default marker value. Perhaps we could return null instead, but in practice this doesn't occur in our codebase so doesn't matter.